internal/connectors/engine/workflow/fetch_trades.go

```diff
@@ -0,0 +1,216 @@
+				tradesWithLegs = append(tradesWithLegs, trade)
+			}
+		}
+
+		// Store all generated payments atomically
+		// Using infiniteRetryContext ensures this will retry until successful,
+		// maintaining consistency between trades and their payments
+		if len(paymentsToCreate) > 0 {
+			err = activities.StoragePaymentsStore(
+				infiniteRetryContext(ctx),
+				paymentsToCreate,
+			)
+			if err != nil {
+				return errors.Wrap(err, "storing payments generated from trades")
+			}
+
+			// STEP 3: Update trades with payment leg references
+			// Now that payments are persisted with their IDs, we can link them back to the trade
+			// This creates the bidirectional relationship: Trade -> Payment (via legs) and
+			// Payment -> Trade (via metadata)
+			for i, trade := range tradesWithLegs {
+				basePaymentIdx := i * 2      // Each trade generates 2 payments
+				quotePaymentIdx := i*2 + 1
+				
+				basePayment := paymentsToCreate[basePaymentIdx]
+				quotePayment := paymentsToCreate[quotePaymentIdx]
+
+				// Build trade legs that reference the payment IDs
+				// Direction semantics: CREDIT = funds coming in, DEBIT = funds going out
+				trade.Legs = []models.TradeLeg{
+					{
+						Role:      models.TRADE_LEG_ROLE_BASE,
+						Direction: func() models.TradeLegDirection {
+							if trade.Side == models.TRADE_SIDE_BUY {
+								return models.TRADE_LEG_DIRECTION_CREDIT // Buying = receiving base asset
+							}
+							return models.TRADE_LEG_DIRECTION_DEBIT // Selling = giving base asset
+						}(),
+						Asset:     trade.Market.BaseAsset,
+						NetAmount: basePayment.Adjustments[0].Amount.String(), // Use first adjustment amount
+						PaymentID: &basePayment.ID,
+						Status:    &basePayment.Status,
+					},
+					{
+						Role:      models.TRADE_LEG_ROLE_QUOTE,
+						Direction: func() models.TradeLegDirection {
+							if trade.Side == models.TRADE_SIDE_BUY {
+								return models.TRADE_LEG_DIRECTION_DEBIT // Buying = spending quote asset
+							}
+							return models.TRADE_LEG_DIRECTION_CREDIT // Selling = receiving quote asset
+						}(),
+						Asset:     trade.Market.QuoteAsset,
+						NetAmount: quotePayment.Adjustments[0].Amount.String(),
+						PaymentID: &quotePayment.ID,
+						Status:    &quotePayment.Status,
+					},
+				}
+
+				// Update the trade in the original slice so events are sent with complete data
+				for j := range trades {
+					if trades[j].ID == trade.ID {
+						trades[j].Legs = trade.Legs
+						break
+					}
+				}
+			}
+
+			// Store trades again with updated leg information
+			// This is an upsert operation, so it's safe to call multiple times (idempotent)
+			if len(tradesWithLegs) > 0 {
+				err = activities.StorageTradesStore(
+					infiniteRetryContext(ctx),
+					trades, // Use full trades slice with updated legs
+				)
+				if err != nil {
+					return errors.Wrap(err, "updating trades with payment legs")
+				}
+			}
+		}
+
+		if len(nextTasks) > 0 {
+			// Logic for next tasks if needed
```

Comment:
You're probably not seeing any issues with this remaining unimplemented because your connector put fetch trades as the last element in the plugin workflow. Other connectors might not have trades being the last element, which will result in the rest of the workflow never executing. Let's handle this properly.